### PR TITLE
promtail: enable structured_metadata convert labels

### PR DESCRIPTION
**Pull request by gonzalesraul**
**What this PR does / why we need it**:

There are several labels that can be extracted by the supports service discovery providers, however the structured metadata wouldn't allow to parse labels to it.

This PR allows the users to update the configuration to read from labels discovered, down below are some of the metadata from a pod, that could be of interested of anyone to have that pushed to Loki without creating several streams due to either high-carnality or proliferation of idle chunks:

```
__address__="10.0.165.166:3100"
__meta_kubernetes_namespace="loki"
__meta_kubernetes_pod_annotation_checksum_config="f008ef39c4f3230c5e153c6e9fde51a678e6f0b1087b3be7d91b8142a03dbe7c"
__meta_kubernetes_pod_annotation_kubectl_kubernetes_io_default_container="loki"
__meta_kubernetes_pod_annotationpresent_checksum_config="true"
__meta_kubernetes_pod_annotationpresent_kubectl_kubernetes_io_default_container="true"
__meta_kubernetes_pod_container_id="containerd://e6aef5c39c3fe806d39ea536b198c8c39cdee01d23141fed91e8fe1573cc5e08"
__meta_kubernetes_pod_container_image="docker.io/grafana/loki:2.9.1"
__meta_kubernetes_pod_container_init="false"
__meta_kubernetes_pod_container_name="loki"
__meta_kubernetes_pod_container_port_name="http-metrics"
__meta_kubernetes_pod_container_port_number="3100"
__meta_kubernetes_pod_container_port_protocol="TCP"
__meta_kubernetes_pod_controller_kind="StatefulSet"
__meta_kubernetes_pod_controller_name="loki-write"
__meta_kubernetes_pod_host_ip="10.249.129.191"
__meta_kubernetes_pod_ip="10.0.165.166"
__meta_kubernetes_pod_label_app_kubernetes_io_component="write"
__meta_kubernetes_pod_label_app_kubernetes_io_instance="loki"
__meta_kubernetes_pod_label_app_kubernetes_io_name="loki"
__meta_kubernetes_pod_label_app_kubernetes_io_part_of="memberlist"
__meta_kubernetes_pod_label_controller_revision_hash="loki-write-75847dc945"
__meta_kubernetes_pod_label_statefulset_kubernetes_io_pod_name="loki-write-1"
__meta_kubernetes_pod_labelpresent_app_kubernetes_io_component="true"
__meta_kubernetes_pod_labelpresent_app_kubernetes_io_instance="true"
__meta_kubernetes_pod_labelpresent_app_kubernetes_io_name="true"
__meta_kubernetes_pod_labelpresent_app_kubernetes_io_part_of="true"
__meta_kubernetes_pod_labelpresent_controller_revision_hash="true"
__meta_kubernetes_pod_labelpresent_statefulset_kubernetes_io_pod_name="true"
__meta_kubernetes_pod_name="loki-write-1"
__meta_kubernetes_pod_node_name="ip-10-249-129-191.ec2.internal"
__meta_kubernetes_pod_phase="Running"
__meta_kubernetes_pod_ready="false"
__meta_kubernetes_pod_uid="b3937321-fe90-4e15-ac94-495c8fdb9202"
```

**Which issue(s) this PR fixes**:
https://github.com/grafana/loki/issues/10766

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [x] Tests updated
- [x] `CHANGELOG.md` updated
  - [x] If the change is worth mentioning in the release notes, add `add-to-release-notes` label
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/setup/upgrade/_index.md`
- [ ] For Helm chart changes bump the Helm chart version in `production/helm/loki/Chart.yaml` and update `production/helm/loki/CHANGELOG.md` and `production/helm/loki/README.md`. [Example PR](https://github.com/grafana/loki/commit/d10549e3ece02120974929894ee333d07755d213)

### CODE REVIEW

File: docs/sources/send-data/promtail/stages/structured_metadata.md [hunk 1]
```diff
@@ -44,10 +46,40 @@ For the given pipeline:
...
+
+Given the following discovered labels below with a log line `sample log`:
+
+|- Discovered label |- Value |- Target label |
+| - | - | - |
+| __meta_kubernetes_pod_host_ip | 127.0.0.1 | pod_host_ip |
+| __meta_kubernetes_pod_uid | b3937321-fe90-4e15-ac94-495c8fdb9202 | pod_uid |
+
+The `structured_metadata` stage would turn the discovered labels `pod_uid` and `pod_host_ip` in key-value pair as a structured metadata to the log line `sample log`
+excluding them out of creating high-cardinality streams.
```
Comment (JStickler) [inline docs/sources/send-data/promtail/stages/structured_metadata.md:85]:
```suggestion
The `structured_metadata` stage would turn the discovered labels `pod_uid` and `pod_host_ip` into key-value pairs as  structured metadata for the log line `sample log` and
exclude them from creating high-cardinality streams.
```

**Review by JStickler — COMMENTED** [review]
[docs team] One wording suggestion (if I'm understanding this correctly, I wasn't clear if there was one key value pair or multiple key value pairs being created in your example).

The engineering team will still need to approve the Go updates.

**Review by JStickler — COMMENTED** [review]
[Docs team] Docs LGTM, the engineering team will still need to review and approve the code changes.

**Review by salvacorts — APPROVED** [review]
This looks great! Thanks for your contribution :) 
I'll share it with other folks so they can review it as well.

**Comment by jplouis** [comment]
Any plans to back port this for 2.9.x? Any way I can help with that?

**Comment by cstyan** [comment]
Generally we only backport bug fixes, security fixes, fixes to newly introduced features, or performance fixes. We're also pausing new features in promtail in general, and all new development will most likely go into the [Grafana agent](https://github.com/grafana/agent).

**Pull request merged**